terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -904,6 +895,26 @@ func (r *ClusterResource) apply(ctx context.Context, data *ClusterResourceModel,
 	return diags
 }
 
+func (r *ClusterResource) getImageVersion(ctx context.Context, data *ClusterResourceModel) (imageAttribute, semver.Semver, diag.Diagnostics) {
+	var image imageAttribute
+	diags := data.Image.As(ctx, &image, basetypes.ObjectAsOptions{})
+	imageSemver, err := semver.New(image.Version)
+	if err != nil {
+		diags.AddAttributeError(
+			path.Root("image").AtName("version"),
+			"Invalid image version",
+			fmt.Sprintf("Parsing image version (%s): %s", image.Version, err))
+		return imageAttribute{}, semver.Semver{}, diags
+	}
+	if err := compatibility.BinaryWith(r.providerData.Version.String(), imageSemver.String()); err != nil {
+		diags.AddAttributeError(
+			path.Root("image").AtName("version"),
+			"Image version incompatible with provider version",
```

Comment:
```suggestion
			"Invalid image version",
```

nit: align this with the error message for an invalid microservice version